pandablocks/utils.py

```diff
@@ -0,0 +1,129 @@
+from typing import Dict, Iterable, List, Sequence, Union, cast
+
+import numpy as np
+import numpy.typing as npt
+
+from pandablocks.responses import TableFieldInfo
+
+UnpackedArray = Union[
+    npt.NDArray[np.int32],
+    npt.NDArray[np.uint8],
+    npt.NDArray[np.uint16],
```

Comment:
Compared to the original code in PandaBlocks-ioc, I can't see where the `uint8` and `uint16` types are ever being created?

Reply:
Ah yep I think I accidentally took it out. I'll also add an `astype(np.uint32)`.

Reply:
We decided there was no benefit to doing the `astype`, it's additional overhead and doesn't give anything extra for the user

Reply:
I think I've missed part of this discussion so I'll need to catch up. From my understanding, we're now deliberately converting all numeric values from PandA into an `np.int32`? So we will never now return a `np.uint8` or `np.uint16`?

And then when we pack data for writing back to PandA we only expect to be passed `np.int32` arrays? Doesn't that mean users could now much more easily write numbers that are too large for PandA, but according to the Python typing they'd be fine?

Reply:
Let's discuss this in person...

Reply:
> We decided there was no benefit to doing the `astype`, it's additional overhead and doesn't give anything extra for the user

If you don't `astype` it won't infer type: it's `int32`, or `uint32` if there's no negatives. Values are the same, and there's a mask in the table code anyway so in the packing and unpacking nothing will go wrong.

Do we still want to leave out the `astype`, given that numpy won't infer `uint8` (I think we were unsure about whether numpy would do this in our discussion yesterday)?

Reply:
I think I'm happy leaving all returned numbers in an `int32`/`uint32`, and deleting the type hints for the other numeric types. The current code doesn't return them (right?) so I don't see the need to add it. The way we pass values to the packing function doesn't depend upon this typing at all anyway.

Reply:
Works for me!